src/hooks/number-validator/number-validator.jsx

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react';
+
+// TODO: Figure out how to write tests for this, write them
+export default function useNumberValidator(input) {
+  const [valid, setValid] = useState(true);
+
+  useEffect(() => {
+    if (input === '') {
+      setValid(true);
+    } else {
+      setValid(!!parseFloat(input));
```

Comment:
This doesn't work with `123123asda1` as an input.

Reply:
Yeah, `parseFloat` wasn't cutting it. There's a whole new regex in place now.